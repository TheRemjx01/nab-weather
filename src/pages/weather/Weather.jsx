import React, { useState} from 'react';
import get from 'lodash/get'
import SearchBar from '../../components/search-bar';
import WeatherDisplay from '../../components/weather-display';
import Title from "../../components/document/Title";
import Hidden from "../../components/hidden";

const Weather = () => {
    const [ selectedLocation, setSelectedLocation ] = useState(null);
    const title = 'Weather App'
    return (
        <>
            <Title text={title}/>
            <h1>{title}</h1>
            <SearchBar onChange={setSelectedLocation} />
            <Hidden when={!selectedLocation}>
                <WeatherDisplay
                 locationLabel={get(selectedLocation, 'label')}
                 locationId={get(selectedLocation, 'value')}
                />
            </Hidden>

        </>
    );
};

export default Weather;