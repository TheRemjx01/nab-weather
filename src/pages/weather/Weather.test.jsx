import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Weather from "./Weather";

describe("<Weather />", () => {
  it("should be able to work correctly", () => {
    const Wrapper = shallow(<Weather />);
    expect(toJson(Wrapper)).toMatchSnapshot();
  });
});
