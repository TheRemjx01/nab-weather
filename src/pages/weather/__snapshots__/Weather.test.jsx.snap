// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weather /> should be able to work correctly 1`] = `
<Fragment>
  <Title
    text="Weather App"
  />
  <h1>
    Weather App
  </h1>
  <SearchBar
    onChange={[Function]}
  />
  <Hidden
    when={true}
  >
    <WeatherDisplay />
  </Hidden>
</Fragment>
`;
